shelley/chain-and-ledger/executable-spec/src/Slot.hs

```diff
@@ -48,3 +53,11 @@ firstSlot = slotFromEpoch
 -- | Hard coded global constant for number of slots per epoch
 slotsPerEpoch :: Natural
 slotsPerEpoch = 100
+
+-- | Convert `Slot` data from Byron to Shelley, there should be a check that
+-- Shelley slots fit into `Word64` used in Byron.
```

Comment:
... but this check should be in the other function, shouldn't it? 

And any reason why we don't have this check in place now?

Also note that when elaborating this into a concrete type we will need to cast an unbounded integer into a `Word64` which might give us some headaches. This was the reason why I restricted the type of `VKeys` (but maybe there are other solutions :man_shrugging: )